flink-yarn/src/test/java/org/apache/flink/yarn/TaskExecutorProcessSpecContainerResourcePriorityAdapterTest.java

```diff
@@ -0,0 +1,109 @@
+ */
+
+package org.apache.flink.yarn;
+
+import org.apache.flink.api.common.resources.CPUResource;
+import org.apache.flink.configuration.MemorySize;
+import org.apache.flink.runtime.clusterframework.TaskExecutorProcessSpec;
+import org.apache.flink.util.TestLogger;
+
+import org.apache.hadoop.yarn.api.records.Priority;
+import org.apache.hadoop.yarn.api.records.Resource;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link TaskExecutorProcessSpecContainerResourcePriorityAdapter}.
+ */
+public class TaskExecutorProcessSpecContainerResourcePriorityAdapterTest extends TestLogger {
+
+	private static final Resource MAX_CONTAINER_RESOURCE = Resource.newInstance(102400, 100);
+
+	private static final TaskExecutorProcessSpec TASK_EXECUTOR_PROCESS_SPEC_1 = new TaskExecutorProcessSpec(
+		new CPUResource(1.0),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100),
+		MemorySize.ofMebiBytes(100));
+
+	private static final TaskExecutorProcessSpec TASK_EXECUTOR_PROCESS_SPEC_2 = new TaskExecutorProcessSpec(
+		new CPUResource(2.0),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200),
+		MemorySize.ofMebiBytes(200));
+
+	@Test
+	public void testGetResource() {
+		final TaskExecutorProcessSpecContainerResourcePriorityAdapter adapter = getAdapter();
+		final Resource resource = adapter.getResource(TASK_EXECUTOR_PROCESS_SPEC_1).get();
+		assertThat(resource.getMemory(), is(TASK_EXECUTOR_PROCESS_SPEC_1.getTotalProcessMemorySize().getMebiBytes()));
+		assertThat(resource.getVirtualCores(), is(TASK_EXECUTOR_PROCESS_SPEC_1.getCpuCores().getValue().intValue()));
+	}
+
+	@Test
+	public void testGetPriority() {
+		final TaskExecutorProcessSpecContainerResourcePriorityAdapter adapter = getAdapter();
+		final Priority priority1 = adapter.getPriority(TASK_EXECUTOR_PROCESS_SPEC_1).get();
+		final Priority priority2 = adapter.getPriority(TASK_EXECUTOR_PROCESS_SPEC_2).get();
+		final Priority priority3 = adapter.getPriority(TASK_EXECUTOR_PROCESS_SPEC_1).get();
+		assertThat(priority1, not(priority2));
+		assertThat(priority1, is(priority3));
+	}
+
+	@Test
+	public void testMaxContainerResource() {
+		final TaskExecutorProcessSpecContainerResourcePriorityAdapter adapter =
+			new TaskExecutorProcessSpecContainerResourcePriorityAdapter(
+				1,
+				Resource.newInstance(100, 1),
+				Collections.emptyMap());
+		assertThat(adapter.getResource(TASK_EXECUTOR_PROCESS_SPEC_2).isPresent(), is(false));
+		assertThat(adapter.getPriority(TASK_EXECUTOR_PROCESS_SPEC_2).isPresent(), is(false));
```

Comment:
It seems we could also use `getAdapter` here. We just need a `TASK_EXECUTOR_PROCESS_SPEC_3` with `CPUResource` 300.